core/numeric/step_spec.rb

```diff
@@ -78,8 +78,8 @@
       1.step(5, "foo").should be_an_instance_of(enumerator_class)
     end
 
-    invalid_step = ruby_version_is("2.1".."2.3") {ArgumentError}
-    invalid_step ||= StandardError # TypeError
+    invalid_step = ruby_version_is("2.0".."2.3") {ArgumentError}
+    invalid_step ||= TypeError
```

Comment:
@nobu Thanks for fixing this!
I changed a bit the style in 8d4cadbe7258f0765abc34b789e0c2d8cf5130d2.
The rationale is I want to avoid as much meta-testing logic as possible, and a bit of duplication is not a problem in specs (if it was too much code we could use `shared examples` but here it does not seems to be worth it).